RecipeApi/Authentication/SameAuthorRequirementHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using RecipeApi.Database;
+
+namespace RecipeApi.Authentication
+{
+    public class SameAuthorRequirementHandler : AuthorizationHandler<SameAuthorRequirement, Recipe>
+    {
+        public SameAuthorRequirementHandler()
+        {
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        SameAuthorRequirement requirement,
+        Recipe recipe)
+        {
+            var userId = context.User.FindFirst(c => c.Type == ClaimTypes.NameIdentifier).Value;
```

Comment:
You can get user name from the token.
https://github.com/technologiesforaccessibility/osmintegrator-backend/blob/0fe312aa147e0e5a086c4885e70d2c49747498db/OsmIntegrator/Controllers/RolesController.cs#L65